### ISSUE TITLE
Professional Audioplayer

### ISSUE THREAD

**Issue by Moments-of-Madness** [pojome/elementor#2858]
Today Elementor include two alternatives for podcasters and musicians. One is Soundcloud which is not really an Elementor developed solution but an embedded player from the service provider Soundcloud (it might suit some but definitely not all), the other is the standard WordPress audio player which is super limited (and additionally the player looks different in the design interface and live).  The player looks different on the backend compared to frontend because a different one is applied to the backend compared to the front. The default WordPress audio player is used for the frontend vs the html5 one on the backend provided by the browser. Naturally, I suggest Elementor fix this but additionally, the html5 player should be extended to include the following functionality:

1. Formats: The player should be able to play both WAV and MP3 files (as the standard-player already does)
2 Style: It should be customizable (so that style, colours etc can be adjusted to a design)
3 Mobile friendly: It should be more mobile friendly than both Soundcloud and the standard player. Soundcloud asks if it should open in the Soundcloud app all the time disturbing the experience and the standard player isn't really made for a mobile format (we need to design the ultimate UX).
4. Resizability: It could be designed to look similar to the Soundcloud player make sure it will work in different sizes (even in a 50px high clean version)
5. Speed control: It should include speed control allowing listeners to adjust play speed 0.5x, 1x, 1.5x and 2x
6. External audio-files: It should be possible to play audio-files hosted outside the WordPress site itself. For podcasters, audio-files are as good as always hosted externally.
7. Download file: If the designer selects this, it should be possible to download the audio-file directly from the player.
8. Song graphics: It should be possible to add a cover for the song (1400x1400 - 3000x3000 px is standard but should be down-scaled to fit) 

For a version 2 in the future, it may be interesting to include playlists for audio-galleries and more control for both user and designer but I suggest that we start with a good basic player. The two options available today are not sustainable and a good player would make Elementor more popular with musicians and podcasters which is a rather big market.

**Comment by bainternet** [comment]
@Moments-of-Madness 

thanks for your suggestions, currently something like developing a full-featured html5 audio player like the one you described is not really on our roadmap and is kept as more of a 3rd party territory.
Unless we see a large demand for it.

As for the native WordPress widget not looking the same (preview vs. front-end) we'll check that.

Again thanks for your suggestions.

**Comment by zebrahosting** [comment]
+1 for this request. There is a nice video option so why not an audio one. Start simpel with a customisable  HTML mp3 player skin.

**Comment by magiceupho** [comment]
+1 and I need the audio player able to play a dynamic URL from an ACF custom filed in order to include that in a template for my custom post type plz.

**Comment by simonrperry** [comment]
Yes - Having the URL pulled in dynamically from ACF is where I'd like things to end up too.

Until then, can someone suggest a way of achieving something like this with some code?

**Comment by simonrperry** [comment]
OK - Here's a quick and dirty solution:

I stored the audio's URL in an ACF text field, then on the Elementor's template I had the ACF field appear to show the text field (URL), but in the Advanced drop down put the following in:
Before: [audio src="
After: "]
and the built-in Wordpress audio widget appeared with the audio file URL pulled from ACF.

Hope this helps people until Elementor supports.

**Comment by SubtleMonsters** [comment]
@bainternet I agree that an extensive solution may be outside Elementor's scope, but it would be nice to at least access a html 5 audio element with a dynamic custom field. I wouldn't think this is much different than the image implementation, is it?

**Comment by alissit** [comment]
> OK - Here's a quick and dirty solution:
> 
> I stored the audio's URL in an ACF text field, then on the Elementor's template I had the ACF field appear to show the text field (URL), but in the Advanced drop down put the following in:
> Before: [audio src="
> After: "]
> and the built-in Wordpress audio widget appeared with the audio file URL pulled from ACF.
> 
> Hope this helps people until Elementor supports.

Thank you!!  I've been trying to find a solution for this for days now!

**Comment by jafdpmoran** [comment]
It may be of interest to the Elementor firmament that Audio Books, Audio for games, Narration for all sorts of enterprise is major, major business. I would concur heartily Moments of Ma at the beginning of this thread. Audio is a major, major market and Elementor is behind the 8 with pro, inventive widgets, both single and playlists. Everything else? Superb,

**Comment by maxjutras** [comment]
Perhaps not the solution you are looking for but Sonaar Themes are very nice niché themes for Music and Podcast website that are 100% compatible with Elementor (they are specifically designed for elementor) 

They are also listed on the official Elementor website as one of the best theme to work with. https://sonaar.io

**Comment by cloudhigh** [comment]
In Elementor text block [audio mp3="https://YOURURL.mp3"][/audio]

**Comment by patperdue** [comment]
+1 Please add this feature in the next release.  With the explosion of podcasting, it is unacceptable that this hasn't yet been included yet.  Thank you.

**Comment by bayslim** [comment]
+1 I need this exact thing. The Sonaar plugin did not work for me. I was able to hook my podcast up to the video widget, but it didn't look right. I was able to use simonrperry's brilliant solution for now, but I'd love to see an audio player I could use with dynamic content in future.

**Comment by 79ho3ein** [comment]
+1

**Comment by jagge72** [comment]
+1

**Comment by calvinhanson** [comment]
I agree there should be an audio widget. I did a workaround as well using what others said, but I used ACF File field so it is easier than copy and pasting the url. 

- then in the Elementor theme builder, I used the "Shortcodes" widget. 
- On the Dynamic dropdown I DON'T select ACF Field, but I select "Custom Post Type" and manually enter the Field Name (you can find this when you are in the ACF area under "Field Name".
- I follow the instructions above with the "before" and "after" in the Shortcodes widget.
- Save and view, it should show the HTML5 player on the front end.

Hope this helps others out there.

**Comment by ayoubkhan558** [comment]
+1
